WIP User Story 1 spacecraft movement

alangomez24 opened the pull request:
Issue #1 

First iteration of our main HTML page. There is a newTurn form that contains the user input fields for movement. The 'Proceed' button calls my spaceship.move() function when clicked.

The move() function changes the location of the ship based off of the direction and distance inputs. It also decreases energy and supplies appropriately as well as checks to see if the ship has left the play area (aka entered a wormhole). Then the move function calls the setData() function, which updates status data that is displayed to the user.

SimonBartonPSU committed:
Create SpaceHunt.html

Made a barebones HTML template to start with

SimonBartonPSU committed:
Moved text, indent formatted

minor text movement, added tabs and added title

halyngoc reviewed (changes requested):
- [x] All input elements in a form should be in one single form element. Example from https://www.w3schools.com/html/html_forms.asp

   ```html
   <form>
     <input type="radio" name="gender" value="male" checked> Male<br>
     <input type="radio" name="gender" value="female"> Female<br>
     <input type="radio" name="gender" value="other"> Other
   </form>
   ```

- [x] Use proper indentation, 2 spaces

- [x] Inconsistency with `name=value` vs `name="value"` (for example [here](https://github.com/halyngoc/CS300-Group3-SpaceHunt/pull/9/files#diff-b571208aa78bb8701815736f96a859aeR41)). I'd say go with the 2nd one cause most tutorials I see do that

- [x] `<label for="elementID">`. See [tutorial here](https://www.w3schools.com/tags/tag_label.asp). So for example [here](https://github.com/halyngoc/CS300-Group3-SpaceHunt/pull/9/commits/a79202e7a117efb04b35a51890cb8d2bcb1b6d3e#diff-b571208aa78bb8701815736f96a859aeR11) either do this

   ```html
   <label for="distance">Distance </label>
   <input type=text name=distance id="distance" value=0>
   ```
   or
   ```html
   <label>Distance <input type=text name=distance value=0></label>
   ```

SimonBartonPSU committed:
Create gameObjects.js

File with some object defintions

SimonBartonPSU committed:
Incremental changes

Formatting, working on move function and spaceship object

SimonBartonPSU commented:
Will work on these fixes and complete movement behavior tomorrow. Good suggestions for sure. 👍

SimonBartonPSU committed:
Minor improvements

Got a little done today, but still not functional.

SimonBartonPSU committed:
Finished first implementation of movement

Fixed indentation in HTML and js files. Got movement to a working point. System status is updated upon movement and spaceship location is set to (75,75) upon wormhole (out of bounds) - this is the non-random wormhole behavior.

SimonBartonPSU commented:
OK I've got my user story to an initial working point. I wanted to get system status info to display with output tags but couldn't figure it out, so the system status displays with input text tags.

SimonBartonPSU committed:
Made max coordinate variable

Max map size is flexible with one spaceship field (maxCoord)

SimonBartonPSU committed:
Removed console.log statements

Removed unnecessary debugging

SimonBartonPSU committed:
HTML simplified and setData uses getElementById

Switched to paragraph tags for current status data. Removed unneccessary HTML labels and input tags. setData funtion now calls getElementById to update the data.

SimonBartonPSU committed:
Fixed semicolon

Added a missing semicolon

SimonBartonPSU committed:
Semicolon

Trying to fix the semicolon error from the bot
